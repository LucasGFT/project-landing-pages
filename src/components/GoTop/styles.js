import styled, { css } from 'styled-components';

export const Container = styled.a`
    ${({ theme }) => css`
        position: fixed;
        display: flex;
        background-color: ${theme.colors.primaryColor};
        color: ${theme.colors.white};
        justify-content: center;
        align-items: center;
        width: 4rem;
        height: 4rem;
        bottom: 2rem;
        right: 2rem;
        z-index: 6;
        /* opacity: 0.7; */
        border: 0.1rem solid ${theme.colors.primaryColor};
        border-radius: 50%;
        text-decoration: none;
    `}
`;
