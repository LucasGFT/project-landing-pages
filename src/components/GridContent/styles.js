import styled, { css } from 'styled-components';

export const Container = styled.div`
    ${({ theme }) => css`
        /* display: flex; */
        /* flex-direction: column; */
        text-align: center;
        /* min-height: 100vh; */
        max-width: 58rem;
        margin: 0 auto;
    `}
`;

export const Html = styled.div`
    ${({ theme }) => css`
        margin: ${theme.spacings.xhuge} 0;
    `}
`;
