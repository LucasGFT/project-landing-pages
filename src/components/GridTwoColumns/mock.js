export default {
  title: 'Grid Two Columns',
  text: `Lorem ipsum dolor sit amet, consectetur adipisicing elit. Vel voluptatem, et fugit dolorum nulla inventore cumque fugiat officiis explicabo, ab a nihil sit animi temporibus excepturi! Atque eos maxime tempora!`,
  srcImg: 'assets/images/javascript.svg',
};
