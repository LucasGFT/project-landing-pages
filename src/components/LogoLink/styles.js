import styled, { css } from 'styled-components';

export const Link = styled.a`
    ${({ theme }) => css`
        all: unset;
        color: ${theme.colors.primaryColor};
        font-weight: 700;
        cursor: pointer;

        > img {
            height: 3rem;
        }
    `}
`;
