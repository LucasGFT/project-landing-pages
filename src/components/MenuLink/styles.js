import styled, { css } from 'styled-components';

export const Link = styled.a`
    ${({ theme }) => css`
        display: block;
        text-decoration: none;
        color: ${theme.colors.primaryColor};
        font-size: ${theme.font.sizes.small};
        padding: ${theme.spacings.small};
        position: relative;

        &::after {
            content: '';
            position: absolute;
            bottom: 0.75rem;
            left: 50%;
            width: 0;
            height: 0.2rem;
            background-color: ${theme.colors.secondaryColor};
            transition: all 300ms ease-in-out;
        }

        &:hover::after {
            left: 25%;
            width: 50%;
        }
    `}

    
`;
