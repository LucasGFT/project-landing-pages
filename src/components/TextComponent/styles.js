import styled, { css } from 'styled-components';

export const Text = styled.div`
    ${({ theme }) => css`
        font-size: ${theme.font.sizes.medium};
    `}
`;
