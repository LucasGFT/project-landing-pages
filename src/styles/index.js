export * from './global-styles';
export * from './theme';
export * from './render-theme';
