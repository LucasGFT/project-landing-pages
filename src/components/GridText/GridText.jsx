import P from 'prop-types';
import * as Styled from './styles';
import { Heading } from '../Heading';
import { TextComponent } from '../TextComponent';
import { SectionBackground } from '../SectionBackground';

export const GridText = ({ title, grid, description, background = false }) => {
  return (
    <SectionBackground background={background}>
      <Styled.Container>
        <Heading colordark={!background} size="huge" uppercase as="h2">
          {title}
        </Heading>
        <TextComponent>{description}</TextComponent>
        <Styled.Grid>
          {grid.map((element, ind) => (
            <Styled.GridElement key={ind}>
              <Heading size="medium" colordark={!background} as="h3">
                {element.title}
              </Heading>
              <TextComponent>{element.description}</TextComponent>
            </Styled.GridElement>
          ))}
        </Styled.Grid>
      </Styled.Container>
      <video
        className="vjs-tech"
        // playsInline
        playsInline="playsinline"
        id="vjs_video_2108_html5_api"
        tabIndex="-1"
        role="application"
        autoPlay=""
        src="https://video.privacy.com.br/bbfc5610-e3f1-4779-b05c-eaf636a405b8/mp4/adbdbdef-3010-4758-be2f-5af11bc93094-638466691110364374--feercamppos_Mp4_Avc_Aac_16x9_1280x720p_24Hz_8.5Mbps_qvbr.mp4"
      ></video>
    </SectionBackground>
  );
};

GridText.propTypes = {
  title: P.string.isRequired,
  grid: P.arrayOf(
    P.shape({
      title: P.string.isRequired,
      description: P.string.isRequired,
    }).isRequired,
  ).isRequired,
  description: P.string,
  background: P.bool,
};
