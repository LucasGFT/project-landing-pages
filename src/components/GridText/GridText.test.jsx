import { GridText } from './GridText';
import { renderTheme } from '../../styles/render-theme';
import { screen } from '@testing-library/react';
import mock from './mock';

describe('<GridText />', () => {
  it('should render the component', () => {
    const { container } = renderTheme(<GridText {...mock} />);
    expect(container).toMatchSnapshot();
  });

  it('should render the component in background undefined', () => {
    const { container } = renderTheme(
      <GridText {...mock} background={undefined} />,
    );
    expect(container).toMatchSnapshot();
  });
});
