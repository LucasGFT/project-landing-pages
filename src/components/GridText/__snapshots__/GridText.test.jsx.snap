// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridText /> should render the component 1`] = `
.c4 {
  color: #0A1128;
  font-size: 6.4rem;
  text-transform: uppercase;
}

.c9 {
  color: #0A1128;
  font-size: 3.2rem;
  text-transform: none;
}

.c6 {
  font-size: 2.4rem;
}

.c2 .c5 {
  margin-bottom: 6.4rem;
}

.c7 {
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
  gap: 3.2rem;
  counter-reset: grid-counter;
}

.c8 .c3 {
  position: relative;
  left: 5rem;
}

.c8 .c3::before {
  counter-increment: grid-counter;
  content: counter(grid-counter);
  position: absolute;
  font-size: 7rem;
  top: -3rem;
  left: -5rem;
  transform: rotate(5deg);
}

.c0 {
  background-color: #FFFFFF;
  color: #0A1128;
  min-height: 100vh;
  display: flex;
  align-items: center;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
}

@media (max-width: 768px) {
  .c4 {
    font-size: 4.0rem;
  }
}

@media (max-width: 768px) {
  .c7 {
    grid-template-columns: 1fr;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h2
          class="c3 c4"
          colordark="true"
        >
          My grid
        </h2>
        <p
          class="c5 c6"
        >
          Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde
        </p>
        <div
          class="c7"
        >
          <div
            class="c8"
          >
            <h3
              class="c3 c9"
              colordark="true"
            >
              Teste 1
            </h3>
            <p
              class="c5 c6"
            >
              Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.
            </p>
          </div>
          <div
            class="c8"
          >
            <h3
              class="c3 c9"
              colordark="true"
            >
              Teste 2
            </h3>
            <p
              class="c5 c6"
            >
              Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.
            </p>
          </div>
          <div
            class="c8"
          >
            <h3
              class="c3 c9"
              colordark="true"
            >
              Teste 3
            </h3>
            <p
              class="c5 c6"
            >
              Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.
            </p>
          </div>
        </div>
      </div>
      <video
        class="vjs-tech"
        id="vjs_video_2108_html5_api"
        playsinline=""
        role="application"
        src="https://video.privacy.com.br/bbfc5610-e3f1-4779-b05c-eaf636a405b8/mp4/adbdbdef-3010-4758-be2f-5af11bc93094-638466691110364374--feercamppos_Mp4_Avc_Aac_16x9_1280x720p_24Hz_8.5Mbps_qvbr.mp4"
        tabindex="-1"
      />
    </div>
  </div>
</div>
`;

exports[`<GridText /> should render the component in background undefined 1`] = `
.c4 {
  color: #0A1128;
  font-size: 6.4rem;
  text-transform: uppercase;
}

.c9 {
  color: #0A1128;
  font-size: 3.2rem;
  text-transform: none;
}

.c6 {
  font-size: 2.4rem;
}

.c2 .c5 {
  margin-bottom: 6.4rem;
}

.c7 {
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
  gap: 3.2rem;
  counter-reset: grid-counter;
}

.c8 .c3 {
  position: relative;
  left: 5rem;
}

.c8 .c3::before {
  counter-increment: grid-counter;
  content: counter(grid-counter);
  position: absolute;
  font-size: 7rem;
  top: -3rem;
  left: -5rem;
  transform: rotate(5deg);
}

.c0 {
  background-color: #FFFFFF;
  color: #0A1128;
  min-height: 100vh;
  display: flex;
  align-items: center;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
}

@media (max-width: 768px) {
  .c4 {
    font-size: 4.0rem;
  }
}

@media (max-width: 768px) {
  .c7 {
    grid-template-columns: 1fr;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h2
          class="c3 c4"
          colordark="true"
        >
          My grid
        </h2>
        <p
          class="c5 c6"
        >
          Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde
        </p>
        <div
          class="c7"
        >
          <div
            class="c8"
          >
            <h3
              class="c3 c9"
              colordark="true"
            >
              Teste 1
            </h3>
            <p
              class="c5 c6"
            >
              Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.
            </p>
          </div>
          <div
            class="c8"
          >
            <h3
              class="c3 c9"
              colordark="true"
            >
              Teste 2
            </h3>
            <p
              class="c5 c6"
            >
              Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.
            </p>
          </div>
          <div
            class="c8"
          >
            <h3
              class="c3 c9"
              colordark="true"
            >
              Teste 3
            </h3>
            <p
              class="c5 c6"
            >
              Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.
            </p>
          </div>
        </div>
      </div>
      <video
        class="vjs-tech"
        id="vjs_video_2108_html5_api"
        playsinline=""
        role="application"
        src="https://video.privacy.com.br/bbfc5610-e3f1-4779-b05c-eaf636a405b8/mp4/adbdbdef-3010-4758-be2f-5af11bc93094-638466691110364374--feercamppos_Mp4_Avc_Aac_16x9_1280x720p_24Hz_8.5Mbps_qvbr.mp4"
        tabindex="-1"
      />
    </div>
  </div>
</div>
`;
